packages/rum-core/src/domain/vital/vitalCollection.spec.ts

```diff
@@ -30,78 +32,112 @@ describe('vitalCollection', () => {
   })
 
   describe('custom duration', () => {
-    describe('createVitalInstance', () => {
-      it('should create duration vital from a vital instance', () => {
+    describe('startDurationVital', () => {
+      it('should create duration vital from a vital reference', () => {
         const cbSpy = jasmine.createSpy()
 
-        const vital = createVitalInstance(cbSpy, { name: 'foo' })
+        const vitalRef = startDurationVital(vitalsState, 'foo')
         clock.tick(100)
-        vital.stop({})
+        stopDurationVital(cbSpy, vitalsState, vitalRef)
 
         expect(cbSpy).toHaveBeenCalledOnceWith(jasmine.objectContaining({ name: 'foo', duration: 100 }))
       })
 
-      it('should not create duration vital without calling `stop` on vital instance', () => {
+      it('should create duration vital from a vital name', () => {
         const cbSpy = jasmine.createSpy()
 
-        createVitalInstance(cbSpy, { name: 'foo' })
+        startDurationVital(vitalsState, 'foo')
+        clock.tick(100)
+        stopDurationVital(cbSpy, vitalsState, 'foo')
 
-        expect(cbSpy).not.toHaveBeenCalled()
+        expect(cbSpy).toHaveBeenCalledOnceWith(jasmine.objectContaining({ name: 'foo', duration: 100 }))
       })
 
-      it('should not create multiple duration vitals by calling "stop" on the same vital instance multiple times', () => {
+      it('should not create multiple duration vitals by calling "stopDurationVital" on the same vital multiple times', () => {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** It would be clearer if you slip this test. One when using refs and another when using names‏

Reply:
*split 😉

Reply:
Done